react/src/components/KeypairResourcePolicySettingModal.tsx

```diff
@@ -0,0 +1,490 @@
+            },
+            {
+              max: 64,
+            },
+            {
+              validator: (_, value) => {
+                if (
+                  !keypairResourcePolicy &&
+                  existingPolicyNames?.includes(value)
+                ) {
+                  return Promise.reject(
+                    t('resourcePolicy.ResourcePolicyNameAlreadyExists'),
+                  );
+                }
+                return Promise.resolve();
+              },
+            },
+          ]}
+        >
+          <Input disabled={!!keypairResourcePolicy} />
+        </Form.Item>
+        <Form.Item label={t('resourcePolicy.ResourcePolicy')} required>
+          <Card>
+            <Row gutter={16}>
+              <Col span={12}>
+                <FormItemWithUnlimited
+                  form={form}
+                  fieldName={['parsedTotalResourceSlots', 'cpu']}
+                  checkboxFieldName="totalResourceSlotsCPUUnlimited"
+                  label={t('session.launcher.CPU')}
+                  fallbackValue={parsedTotalResourceSlots?.cpu}
+                  render={(disabled) => (
+                    <InputNumber
+                      min={0}
+                      max={512}
+                      disabled={disabled}
+                      addonAfter={t('session.launcher.Core')}
+                    />
+                  )}
+                />
+              </Col>
+              <Col span={12}>
+                <FormItemWithUnlimited
+                  form={form}
+                  fieldName={['parsedTotalResourceSlots', 'mem']}
+                  checkboxFieldName="totalResourceSlotsMEMUnlimited"
+                  label={t('session.launcher.Memory')}
+                  fallbackValue={parsedTotalResourceSlots?.mem}
+                  render={(disabled) => (
+                    <DynamicUnitInputNumber disabled={disabled} />
+                  )}
+                />
+              </Col>
+            </Row>
+            <Row gutter={16}>
+              <Col span={12}>
+                <FormItemWithUnlimited
+                  form={form}
+                  fieldName={['parsedTotalResourceSlots', 'cuda.device']}
+                  checkboxFieldName="totalResourceSlotsGPUUnlimited"
+                  label={t('session.GPU')}
+                  fallbackValue={parsedTotalResourceSlots?.['cuda.device']}
+                  render={(disabled) => (
+                    <InputNumber
+                      min={0}
+                      max={64}
+                      disabled={disabled}
+                      addonAfter={t('session.GPU')}
+                    />
+                  )}
+                />
+              </Col>
+              <Col span={12}>
+                <FormItemWithUnlimited
+                  form={form}
+                  fieldName={['parsedTotalResourceSlots', 'cuda.shares']}
+                  checkboxFieldName="totalResourceSlotsFGPUUnlimited"
+                  label={t('session.fGPU')}
```

Comment:
For the `fGPU` label, please utilize the `useResourceSlotsDetails` hook instead of using a hardcoded string or an i18n item.